client/src/rematch/models.js

```diff
@@ -258,5 +258,43 @@ export const testEnvironment = {
       state.draggedCubic = false;
       return state;
     },
+    addPoint: (state, payload) => {
+      const { coords } = payload;
+      const { activeLine } = state;
+      const vectorLength = state.lines[activeLine].vector.length - 1;
```

Comment:
I am confused. Why are you substracting the length by 1?

Reply:
Good catch, that was the wrong variable name. It should have been something like `lastPointIndex`. It is used when finding the coordinates of the last point. 

Ill fix that.